JS/main.js

```diff
@@ -0,0 +1,574 @@
+
+The function is used in the startDrawing function in order for the user to be able to start drawing.
+
+*/
+
+function draw(e) {
+    if(drawRectangle.checked) {
+        return;
+    };
+
+    if(drawEllipse.checked) {
+        return;
+    };
+
+    if(drawTriangle.checked) {
+        return;
+    };
+
+    if (!drawing) {
+        return;
+    };
+
+    let { x, y } = getMousePos(canvas, e);
+
+    context.lineTo(x, y);
+    context.stroke();
+};
+
+canvas.addEventListener("mousemove", draw);
+
+/* Draw Square
+
+Checks if the drawRectangle HTML Radio Input is checked;
+
+If drawing is set to false, the function exits;
+
+Sets the x and y to the current mouse possition;
+
+Gets a width and a height for the current shape;
+
+Draw the rectangle from the initial mouse position to the latest one;
+
+If the solidSelect is checked, the shape will be filled;
+
+Event is called in the draw function.
+
+*/
+
+function drawSquare(e) {
+    if (!drawing) {
+        return;
+    }
+    
+    let { x, y } = getMousePos(canvas, e);
+
+    let width = x - rectStart.x;
+    let height = y - rectStart.y;
+
+    context.strokeRect(rectStart.x, rectStart.y, width, height);
+
+    if (solidSelect.checked) {
+        context.fillRect(rectStart.x, rectStart.y, width, height);
+    };
+};
+
+/* Draw Ellipse
+
+Checks if the drawEllipse HTML Radio Input is checked;
+
+If drawing is set to false, the function exits;
+
+Sets the x and y to the current mouse possition;
+
+Gets a width and a height for the current shape;
+
+Gets the center by divinding the initial mouse on both axis to 2;  
+
+Calculates the radius for both axis;
+
+Draws the ellipse from the initial mouse position to the latest one.
+
+If the solidSelect is checked, the shape will be filled;
+
+Event is called in the draw function.
+
+*/
+
+function drawCircle(e) {
+    if (!drawing) {
+        return;
+    };
+
+    let { x, y } = getMousePos(canvas, e);
+
+    let width = x - rectStart.x;
+    let height = y - rectStart.y;
+
+    let centerX = rectStart.x + width / 2;
+    let centerY = rectStart.y + height / 2;
+    let radiusX = Math.abs(width / 2);
+    let radiusY = Math.abs(height / 2);
+
+    context.beginPath();
+    context.ellipse(centerX, centerY, radiusX, radiusY, 0, 0, 2 * Math.PI);
+    context.stroke();
+
+    if (solidSelect.checked) {
+        context.fill();
+    };
+};
+
+/* Draw Triangle
+
+Checks if the drawTriangle HTML Radio Input is checked;
+
+If drawing is set to false, the function exits;
+
+Sets the x and y to the current mouse possition;
+
+Calculates the lenghts of each side so that the triangle is more or less equilateral;
+
+Draws the ellipse from the initial mouse position to the latest one.
+
+If the solidSelect is checked, the shape will be filled;
+
+Event is called in the draw function.
+
+*/
+
+function drawTriangl(e) {
+    if (!drawing) {
+        return;
+    };
+
+    let { x, y } = getMousePos(canvas, e);
```

Comment:
Văd că nu modifici valorile x și y. Ai putea să le faci constante.

```suggestion
    const { x, y } = getMousePos(canvas, e);
```

Reply:
Am modificat la forme.